olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -681,5 +749,59 @@ std::string DefaultCacheImpl::GetExpiryKey(const std::string& key) const {
   return CreateExpiryKey(key);
 }
 
+bool DefaultCacheImpl::Protect(const DefaultCache::KeyListType& keys) {
+  std::lock_guard<std::mutex> lock(cache_lock_);
+  auto start = std::chrono::steady_clock::now();
+  protected_keys_.Protect(keys, [&](const std::string& key) {
+    if (!RemoveKeyLru(key)) {
+      RemoveKeysWithPrefixLru(key);
+    }
+  });
+  if (memory_cache_) {
+    memory_cache_->Clear();
+  }
+
+  OLP_SDK_LOG_INFO_F(kLogTag,
+                     "Protect, time=%" PRId64 " ms, added keys size=%" PRIu64
+                     ", total size=%" PRIu64,
+                     GetElapsedTime(start),
+                     static_cast<std::uint64_t>(keys.size()),
+                     protected_keys_.Count());
+  return true;
+}
+
+bool DefaultCacheImpl::Release(const DefaultCache::KeyListType& keys) {
+  std::lock_guard<std::mutex> lock(cache_lock_);
+  auto start = std::chrono::steady_clock::now();
+  auto result = protected_keys_.Release(keys);
+
+  if (memory_cache_) {
+    memory_cache_->Clear();
+  }
+  if (mutable_cache_) {
+    auto it = mutable_cache_->NewIterator(leveldb::ReadOptions());
+
+    for (it->SeekToFirst(); it->Valid(); it->Next()) {
+      AddKeyToLru(it->key().ToString(), it->value());
+    }
+  }
```

Comment:
Why are you adding each key to LRU again? You should only add released keys, resp. keys that have the same prefix as the released key.